callbacks.go

```diff
@@ -136,6 +152,105 @@ func (exec *sidecarExecutor) LaunchTask(driver executor.ExecutorDriver, taskInfo
 	go exec.monitorTask(cntnr.ID[:12], taskInfo)
 }
 
+// getMasterHostname talks to the local worker endpoint and discovers the
+// Mesos master hostname.
+func (exec *sidecarExecutor) getMasterHostname() (string, error) {
+	localEndpoint := "http://" + os.Getenv("MESOS_AGENT_ENDPOINT")
+	if len(localEndpoint) < 8 { // Did we get anything in the env var?
+		return "", fmt.Errorf("Can't get MESOS_AGENT_ENDPOINT from env! Won't provide Sidecar seeds.")
+	}
+
+	localStruct := struct {
+		MasterHostname string `json:"master_hostname"`
+	}{}
+
+	// Let's find out the Mesos master's hostname
+	resp, err := exec.fetcher.Get(localEndpoint)
+	defer func() {
+		if resp != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	if err != nil {
+		return "", fmt.Errorf("Unable to fetch Mesos master info from worker endpoint: %s", err)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("Error reading response body from Mesos worker! '%s'", err)
+	}
+
+	err = json.Unmarshal(body, &localStruct)
+	if err != nil {
+		return "", fmt.Errorf("Error parsing response body from Mesos worker! '%s'", err)
+	}
+
+	return localStruct.MasterHostname, nil
+}
+
+// Used only in getWorkerHostnames
```

Comment:
then why not declare it in the body of the function?

Reply:
haha, I guess that works now :)